WooCommerce/WooCommerceTests/ViewRelated/Shipping Label/WooShipping Create Shipping Labels/WooShippingOriginAddressListViewModelTests.swift

```diff
@@ -0,0 +1,72 @@
+import XCTest
+@testable import WooCommerce
+import Yosemite
+
+final class WooShippingOriginAddressListViewModelTests: XCTestCase {
+
+    func test_it_inits_with_expected_addresses() {
+        // Given
+        let selectedAddressID = "1"
+        let addresses = [WooShippingOriginAddress.fake().copy(id: selectedAddressID), WooShippingOriginAddress.fake()]
+
+        // When
+        let viewModel = WooShippingOriginAddressListViewModel(addresses: addresses, selectedAddressID: selectedAddressID)
+
+        // Then
+        XCTAssertEqual(viewModel.addresses, addresses)
+        XCTAssertEqual(viewModel.selectedAddressID, selectedAddressID)
+    }
+
+    func test_it_defaults_to_selecting_default_address() {
+        // Given
+        let defaultAddress = WooShippingOriginAddress.fake().copy(defaultAddress: true)
+        let addresses = [defaultAddress, WooShippingOriginAddress.fake()]
+
+        // When
+        let viewModel = WooShippingOriginAddressListViewModel(addresses: addresses, selectedAddressID: nil)
```

Comment:
When I saw this line I was expecting that `viewModel.selectedAddressID` will be `nil`, but when I looked at the `init` method of `WooShippingOriginAddressListViewModel` I saw that we have `self.selectedAddressID = selectedAddressID ?? addresses.first(where: { $0.defaultAddress })?.id ?? ""`.

I am wondering if there is a way to inform with better naming or some docs/comments that we use `selectedAddressID` only if it is not nil?

Reply:
> When I saw this line I was expecting that viewModel.selectedAddressID will be nil

That's a good point. I updated this in [f620425](https://github.com/woocommerce/woocommerce-ios/pull/14744/commits/f62042571fbe8d1f78216c7939a5ac31db5ec2ab) to set `selectedAddressID` directly based on what's provided, since that's more expected behavior.

We'll want to pre-select the default origin address for the shipping flow, but we can do that when we initialize the view model.